### DISCUSSION TITLE
using custom css for home page

### DISCUSSION THREAD

**Discussion by Gitranju** [knowledgefutures/pubpub#2673] [General]
Hi all,

I am trying to modify the home page of our pubpub, am I right in thinking the custom CSS that I can use is only applicable to body section within html block? I feel that when it comes to header and footer image/logo we have limited maneuverability . https://ranju.pubpub.org/ is my test site I was able to use the css within home page using custom css and html block but looks like for header and footer I am limited to the options set within puppub ?

Regards,
Ranju

**Comment by tefkah** [comment]
No, you can also modify the header and footer. 
See https://journal.trialanderror.org, if you hover over the logo in the top-right it shows the journal name appearing. I did that using custom CSS.

Some things are more difficult to change in the header/footer though, like the background color which is set using `style=#..` on the HTML itself, so you gotta use some `!important`s here and there

**Comment by Gitranju** [comment]
Thanks for replying to my query ,  Thomas! And would I be correct in thinking that PubPub pages do not support JavaScript?


Best regards
Ranju

Ranju Upadhyay Rai
Programmer @FullStackFem Project and MU Library
Maynooth University

Ollscoil Mhá Nuad, Maigh Nuad, Co. Chill Dara, Éire, W23 VP22
Maynooth University, Maynooth, Co. Kildare, Ireland, W23 VP22
[cid:b3f301df-0387-4d27-9e4a-cd47516423f8]
T: +353 1 708 3378  M: +353 87 98 43811
***@***.******@***.***> W: www.maynoothuniversity.ie/library<http://www.maynoothuniversity.ie/>


All personal data received by the Library will be held safely and securely in compliance with the EU General Data Protection Regulation (GDPR) and the Data Protection Act (Ireland) 2018. The Library may retain personal data for operational, statistical and archival purposes. For further information please consult the University's Data Protection Policy: www.maynoothuniversity.ie/data-protection<http://www.maynoothuniversity.ie/data-protection> or contact the MU Data Protection Officer, at ***@***.******@***.***>.



An t-eolas pearsanta ar fad a fhaigheann an Leabharlann bíonn sé slán agus sabháilte faoina coimirce ar mhaithe le Rialachán Ginearálta maidir le Cosaint Sonraí an Aontais Eorpaigh agus leis an Acht um Chosaint Sonraí (Éire) 2018 a chomhlíonadh. D’fhéadfadh an Leabharlann eolas pearsanta a choimead ar chúiseanna oibríochta, staitistiúil agus cartlainne. Le haghaidh tuilleadh eolais, téigh i gcomhairle le Polasaí  na hOllscoile um Chosaint Sonraí: www.maynoothuniversity.ie/data-protection<http://www.maynoothuniversity.ie/data-protection> nó téigh i dteagmháil le hOifigeach Cosanta Sonraí na hOllscoile ag ***@***.******@***.***>.


________________________________
From: Thomas F. K. Jorna ***@***.***>
Sent: Thursday 18 May 2023 11:39
To: pubpub/pubpub ***@***.***>
Cc: Ranju Upadhyay ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [pubpub/pubpub] using custom css for home page (Discussion #2673)

*Warning*

This email originated from outside of Maynooth University's Mail System. Do not reply, click links or open attachments unless you recognise the sender and know the content is safe.

No, you can also modify the header and footer.
See https://journal.trialanderror.org<https://journal.trialanderror.org/>, if you hover over the logo in the top-right it shows the journal name appearing. I did that using custom CSS.

Some things are more difficult to change in the header/footer though, like the background color which is set using style=#.. on the HTML itself, so you gotta use some !importants here and there

**Reply by tefkah** [reply]
They do support Javascript, but you cannot add your own through e.g. `<script>` tags in an HTML component, they get filtered out.
AFAIK there are a few communities that have access to custom JS, but it's not generally available.

**Comment by Gitranju** [comment]
Thanks Thomas. In my layout I am trying to align the logo image outside of banner(https://ranju.pubpub.org/), any idea how I might be able to do that!


Best regards
Ranju

Ranju Upadhyay Rai
Programmer @FullStackFem Project and MU Library
Maynooth University

Ollscoil Mhá Nuad, Maigh Nuad, Co. Chill Dara, Éire, W23 VP22
Maynooth University, Maynooth, Co. Kildare, Ireland, W23 VP22
[cid:06b0ae89-4e00-46a6-bb7b-5e1c7a7b2f53]
T: +353 1 708 3378  M: +353 87 98 43811
***@***.******@***.***> W: www.maynoothuniversity.ie/library<http://www.maynoothuniversity.ie/>


All personal data received by the Library will be held safely and securely in compliance with the EU General Data Protection Regulation (GDPR) and the Data Protection Act (Ireland) 2018. The Library may retain personal data for operational, statistical and archival purposes. For further information please consult the University's Data Protection Policy: www.maynoothuniversity.ie/data-protection<http://www.maynoothuniversity.ie/data-protection> or contact the MU Data Protection Officer, at ***@***.******@***.***>.



An t-eolas pearsanta ar fad a fhaigheann an Leabharlann bíonn sé slán agus sabháilte faoina coimirce ar mhaithe le Rialachán Ginearálta maidir le Cosaint Sonraí an Aontais Eorpaigh agus leis an Acht um Chosaint Sonraí (Éire) 2018 a chomhlíonadh. D’fhéadfadh an Leabharlann eolas pearsanta a choimead ar chúiseanna oibríochta, staitistiúil agus cartlainne. Le haghaidh tuilleadh eolais, téigh i gcomhairle le Polasaí  na hOllscoile um Chosaint Sonraí: www.maynoothuniversity.ie/data-protection<http://www.maynoothuniversity.ie/data-protection> nó téigh i dteagmháil le hOifigeach Cosanta Sonraí na hOllscoile ag ***@***.******@***.***>.


________________________________
From: Thomas F. K. Jorna ***@***.***>
Sent: Tuesday 23 May 2023 09:13
To: pubpub/pubpub ***@***.***>
Cc: Ranju Upadhyay ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [pubpub/pubpub] using custom css for home page (Discussion #2673)

*Warning*

This email originated from outside of Maynooth University's Mail System. Do not reply, click links or open attachments unless you recognise the sender and know the content is safe.

They do support Javascript, but you cannot add your own through e.g. <script> tags in an HTML component, they get filtered out.
AFAIK there are a few communities that have access to custom JS, but it's not generally available.

**Reply by Gitranju** [reply]
in other words , top left side of the page but outside of the green banner that say "welcome......"!!!

**Reply by tefkah** [reply]
i think it will be difficult to align those, as the div housing the logo also houses the Dasboard, Login etc buttons. So aligning that with the header would require you to deal with those buttons in some way, or really precisely make the header title overlap with that div in the middle, which I would not recommend.

So I don't think there's a great solution here! I've had some similar issues (I'd like to put the navigation menu on top) but also gave up on that eventually.
